File: ParkingProjectAPI/ParkingProjectAPI/Models/Reservation.cs
using System;
using System.Collections.Generic;

namespace ParkingProjectAPI.Models
{
    public partial class Reservation
    {
        public int Id { get; set; }
        public int ParkingId { get; set; }
        public int UserId { get; set; }
        public int SpotId { get; set; }
        public string Date { get; set; } = null!;
        public string CheckinTime { get; set; } = null!;
        public string CheckoutTime { get; set; } = null!;
        public int? VehicleId { get; set; }
        public double TotalRate { get; set; }

        public virtual Parking Parking { get; set; } = null!;
        public virtual Spot Spot { get; set; } = null!;
        public virtual User User { get; set; } = null!;
        public virtual Vehicle? Vehicle { get; set; }
    }
}

File: ApiBookings/src/main/java/com/example/apibookings/domain/Reservation.java
package com.example.apibookings.domain;

//import org.springframework.data.repository.query.Param;
import javax.persistence.*;

@Entity
@Table(name = "Reservation")
public class Reservation {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)

    private int parkingId;
    private int userId;
    private int spotId;
    private int vehicleId;
    private String date;
    private String CheckinTime;
    private String CheckoutTime;
    private float totalRate;

    public Reservation(){

    }

    public Reservation(int parkingId, int userId, int spotId, int vehicleId, String date, String CheckinTime, String CheckoutTime, float totalRate) {
        this.parkingId = parkingId;
        this.userId = userId;
        this.spotId = spotId;
        this.vehicleId = vehicleId;
        this.date = date;
        this.CheckinTime = CheckinTime;
        this.CheckoutTime = CheckoutTime;
        this.totalRate = totalRate;
    }


}
